### DISCUSSION TITLE
Possible to define routes declaratively with <Route>?

### DISCUSSION THREAD

**Discussion by MiguelNiblock** [remix-run/remix#3323] [Q&A] [answered]
The docs say:
https://remix.run/docs/en/v1/guides/migrating-react-router-app#creating-your-routes

>Over time you'll want to migrate the routes rendered by React Router's <Route> components into their own route files.

and 
>Once you create this file, you can delete the <Route> component from your App. After all of your routes have been migrated you can delete <Routes> and ultimately all of the code in old-app.

This suggests the two routing methods (declarative and file-based) may coexist in remix?

**Comment by kiliman** [comment, marked as answer]
Remix routes are defined during build time. It uses file based routing by default but you can define your own routes using *remix.config*. Although it is code based you could define a file (e.g. JSON) and write a function that will build routes based on the file. 

https://remix.run/docs/en/v1/api/conventions#routes
